compaction/compaction_manager.cc

```diff
@@ -550,6 +550,14 @@ sstables::shared_sstable compaction_manager::sstables_task::consume_sstable() {
     return sst;
 }
 
+future<semaphore_units<named_semaphore_exception_factory>> compaction_manager::task::get_units(named_semaphore& sem, size_t units) {
+    return seastar::get_units(sem, units, _compaction_data.abort).handle_exception_type([this] (const semaphore_aborted& e) {
+        auto s = _compacting_table->schema();
```

Comment:
Is it obvious that _compacting_table is always set when we get here?
I'm not sure exactly in what times can we get this abort exception... (and I see below there are several different cases we use this get_units() thing).

Reply:
_compacting_table must not go away while any compaction_manager task refers to it.
We stop all such tasks and wait on them before removing the destroying the instance.

In this case the semaphores are aborted on shutdown when the stop_signal requests abort.